### PR TITLE
Allow Sodium to store block entity information about the shadow pass

### PR DESCRIPTION
**Pull request by IMS212**
This will require a small change to Sodium as well.

### CODE REVIEW

File: src/main/java/net/coderbot/iris/mixin/shadows/MixinPreventRebuildNearInShadowPass.java [hunk 1]
```diff
@@ -28,4 +36,14 @@
...
 	}
+
+	@Inject(method = "setupTerrain", at = @At("TAIL"))
+	private void iris$copyBlockEntities(Camera camera, Frustum frustum, boolean hasForcedFrustum, int frame, boolean spectator, CallbackInfo ci) {
+		ShadowRenderer.visibleBlockEntities.clear();
+		for (WorldRenderer.ChunkInfo chunk : this.visibleChunks) {
+			for (BlockEntity entity : ((ChunkInfoAccessor) chunk).getChunk().getData().getBlockEntities()) {
+				ShadowRenderer.visibleBlockEntities.add(entity);
+			}
+		}
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/mixin/shadows/MixinPreventRebuildNearInShadowPass.java:47]: This should check whether the shadow pass is active before computing visibility info.

**Review by coderbot16 — CHANGES_REQUESTED** [review]

**Comment by coderbot16** [comment]
I think the ShadowRenderer.visibleBlockEntities list should be set to null at the end of shadow rendering to ensure we aren't holding on to (and leaking) a bunch of block entities if the user switches away from a shadowmapped shader pack. The list can be recreated each frame perhaps instead.

**Comment by IMS212** [comment]
For some unknown reason, doing a check for if the shadow renderer is active seems to cause no block entity shadows to render at all.

**Comment by coderbot16** [comment]
That's because the injection is at TAIL, but in the shadow pass setupTerrain is cancelled before rebuildNear. Might as well move the code to the pre-rebuildNear injection I guess.

**Comment by IMS212** [comment]
Makes sense, i'll attempt a workaround for this.

File: src/main/java/net/coderbot/iris/mixin/shadows/MixinPreventRebuildNearInShadowPass.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/net/coderbot/iris/mixin/shadows/MixinPreventRebuildNearInShadowPass.java]: Fixed

**Comment by IMS212** [comment]
This should be fixed.

**Review by coderbot16 — APPROVED** [review]
Thanks!

**Pull request merged**